src/CC2Edit/TileInspector.cpp

```diff
@@ -323,3 +355,76 @@ cc2::Tile* TileInspector::tileLayer(int index)
         tile = tile->lower();
     return tile;
 }
+
+void TileInspector::createLayerAbove() {
+    int layer = m_layers->currentRow();
+    cc2::Tile* tile = tileLayer(layer);
+
+    cc2::Tile newTile = cc2::Tile(cc2::Tile::Chip);
+    *newTile.lower() = *tile;
+    if (layer == 0) m_tile = newTile;

```

Comment:
Style nit: Don't put code on the same line as an `if`/`else` statement...

```suggestion
    if (layer == 0)
        m_tile = std::move(newTile);
```